asim: specify node capacity in cores, not nanos/sec

angeladietz opened the pull request:
Previously, the datadriven asim tests expressed node capacity in nanoseconds/core which are difficult to read due to lots of zeros. Now, the node capacity is expressed in cores (float) which is much more readable and less prone to user error. For example, `node_cpu_rate_capacity=8000000000` is now expressed as `node_cpu_cores=8`.

Fixes #156845
Epic: CRDB-55052
Release note: none.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/158144)
<!-- Reviewable:end -->

angeladietz committed:
asim: specify node capacity in cores, not nanos/sec

Previously, the datadriven asim tests expressed node capacity in
nanoseconds/core which are difficult to read due to lots of zeros.
Now, the node capacity is expressed in cores (float) which is much more
readable and less prone to user error. For example,
`node_cpu_rate_capacity=8000000000` is now expressed as `node_cpu_cores=8`.

Fixes #156845
Epic: CRDB-55052
Release note: none.

wenyihu6 reviewed (approved):
I think only the input, so this should be enough. 
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/158144#-:-OeZtEHq5dh_lT8MNEjK:b-z7yqdn)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained

___
*[`pkg/kv/kvserver/asim/state/node_cpu_cores_test.go` line 14 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/158144#-OeZsxQH1bG3IzeqZ_Z-:-OeZsxQH1bG3IzeqZ_Z0:bds3i29) ([raw file](https://github.com/cockroachdb/cockroach/blob/b418f284ce7a005cd1e6fcd6899f35b844377be9/pkg/kv/kvserver/asim/state/node_cpu_cores_test.go#L14)):*
> ```Go
> )
> 
> func TestNodeCPUCores_ToRateCapacityNanos(t *testing.T) {
> ```

Nice test!


<!-- Sent from Reviewable.io -->

tbg reviewed (approved):
Nice!

> @tbg reviewed 15 of 15 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/158144#-:-Oeadkuw5M7LrkmLpsK5:b-83sjoh)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @angeladietz)

___
*[`pkg/kv/kvserver/asim/config/settings.go` line 36 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/158144#-OeadKDgDfuYWDh1UcHX:-OeadKDgDfuYWDh1UcHY:b-sj3l6y) ([raw file](https://github.com/cockroachdb/cockroach/blob/b418f284ce7a005cd1e6fcd6899f35b844377be9/pkg/kv/kvserver/asim/config/settings.go#L36)):*
> ```Go
> const (
> 	DefaultNodeCPUCores                   = 8.0        // 8 vcpus
> 	DefaultNodeCPURateCapacityNanos       = 8 * 1e9    // 8 vcpus
> ```

nit: make this `DefaultNodeCPUCores * 1e9`.


<!-- Sent from Reviewable.io -->
